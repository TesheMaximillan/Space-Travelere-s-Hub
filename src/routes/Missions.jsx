import React from 'react';
import { useSelector } from 'react-redux';
import Mission from '../components/missions/Mission';
import Thead from '../components/missions/Thead';
import styles from './Missions.module.scss';

function Missions() {
  const missions = useSelector((state) => state.missionReducer);

  const { table } = styles;

  return (
    <table className={table}>
      <thead>
        <Thead />
      </thead>
      <tbody>
        {missions && missions.map((mission) => (
          <Mission key={mission.mission_id} mission={mission} />
        ))}
      </tbody>
      <tfoot />
    </table>
  );
}

export default Missions;
