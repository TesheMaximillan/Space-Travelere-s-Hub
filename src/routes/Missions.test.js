import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Missions from './Missions';
import renderContext from '../components/storeMock';

describe('Myprofile component', () => {
  let component;

  beforeEach(() => {
    component = renderer.create(renderContext(<Missions />));
  });

  it('should render with given state from redux store', () => {
    // Assert
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render Myprofile', () => {
    // Arrange
    render(renderContext(<Missions />));

    // Assert
    expect(screen.getByText(/Mission/i)).toBeInTheDocument();
  });
});
