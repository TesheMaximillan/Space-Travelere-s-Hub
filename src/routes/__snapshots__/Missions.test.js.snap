// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Myprofile component should render with given state from redux store 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
      <th>
         
      </th>
    </tr>
  </thead>
  <tbody />
  <tfoot />
</table>
`;
